src/v2/Apps/Artist/Components/ArtistMetaCanonicalLink.tsx

```diff
@@ -6,27 +6,43 @@ import { ArtistMetaCanonicalLink_artist } from "v2/__generated__/ArtistMetaCanon
 import { hasSections as showMarketInsights } from "v2/Apps/Artist/Components/MarketInsights/MarketInsights"
 import { hasOverviewContent } from "v2/Apps/Artist/Components/NavigationTabs"
 
-export const canonicalPath = (artist: ArtistMetaCanonicalLink_artist) => {
+export const computeCanonicalPath = (
+  artist: ArtistMetaCanonicalLink_artist,
+  path: string
+) => {
+  const basePath = `/artist/${artist.slug}`
+  const pathParts = [basePath]
+
+  const isConsignPage = path === "consign"
+
   const hasArtistInsights =
     showMarketInsights(artist) ||
     (artist.insights && artist.insights.length > 0)
 
   const hasArtistContent = hasOverviewContent(artist)
   const canShowOverview = hasArtistInsights || hasArtistContent
 
-  return canShowOverview
-    ? `/artist/${artist.slug}`
-    : `/artist/${artist.slug}/works-for-sale`
+  if (isConsignPage) {
+    pathParts.push("/consign")
+  } else if (!canShowOverview) {
+    pathParts.push("/works-for-sale")
+  }
+
+  return pathParts.join("")
 }
 
 export type ArtistMetaCanonicalLinkProps = {
   artist: ArtistMetaCanonicalLink_artist
+  path: string
 }
 
 export const ArtistMetaCanonicalLink: React.FC<ArtistMetaCanonicalLinkProps> = ({
   artist,
+  path,
```

Comment:
No need to pass props as [we can use our hook](https://github.com/artsy/force/blob/master/src/v2/Artsy/Router/useRouter.tsx)